packages/rum-core/src/domain/tracing/sampler.ts

```diff
@@ -0,0 +1,19 @@
+import { performDraw } from '@datadog/browser-core'
+import type { TraceIdentifier } from './identifier'
+
+export function isTraceSampled(identifier: TraceIdentifier, sampleRate: number) {
+  // Shortcuts for common cases. This is not strictly necessary, but it makes the code faster for
+  // customers willing to ingest all traces.
+  if (sampleRate === 100) {
+    return true
+  }
+  // TODO: remove condition when all browser we support have BigInt support
+  if (typeof identifier === 'bigint') {
+    const knuthFactor = BigInt('1111111111111111111')
+    const twoPow64 = BigInt('0x10000000000000000') // 2n ** 64n
+    const hash = (identifier * knuthFactor) % twoPow64
+    return Number(hash) < (sampleRate / 100) * Number(twoPow64)
+  }
+  // For simplicity, we don't use deterministic sampling for browser that don't support BigInt
+  return performDraw(sampleRate)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏ I would use an early return for the non-BigInt branch

```suggestion
  // For simplicity, we don't use deterministic sampling for browser that don't support BigInt
  // TODO: remove condition when all browser we support have BigInt support
  if (typeof identifier !== 'bigint') {
    return performDraw(sampleRate)
  }
  
  const knuthFactor = BigInt('1111111111111111111')
  const twoPow64 = BigInt('0x10000000000000000') // 2n ** 64n
  const hash = (identifier * knuthFactor) % twoPow64
  return Number(hash) < (sampleRate / 100) * Number(twoPow64)
```